ooni/scripts/oonideckgen.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import print_function
+
+import errno
+import os
+import sys
+
+from twisted.internet import defer, task
+from twisted.python import usage
+
+from ooni.otime import prettyDateNowUTC
+from ooni import errors
+from ooni.geoip import probe_ip
+from ooni.resources import check_for_update
+from ooni.settings import config
+from ooni.deck import NGDeck
+
+__version__ = "1.0.0"
+
+class Options(usage.Options):
+    synopsis = """%s [options]
+    """ % sys.argv[0]
+
+    optParameters = [
+        ["country-code", "c", None,
+         "Specify the two letter country code for which we should "
+         "generate the deck."],
+        ["collector", None, None, "Specify a custom collector to use when "
+                                  "submitting reports"],
+        ["bouncer", None, None, "Specify a custom bouncer to use"],
+        ["output", "o", None,
+         "Specify the path where we should be writing the deck to."]
+    ]
+
+    def opt_version(self):
+        print("oonideckgen version: " % __version__)
+        sys.exit(0)
+
+def generate_deck(options):
+
+    deck_data = {
+        "name": "Default ooniprobe deck",
+        "description": "Default ooniprobe deck generated on {0}".format(
+                       prettyDateNowUTC()),
+        "schedule": "@daily",
+        "tasks": [
+            {
+                "ooni": {
+                    "test_name": "http_invalid_request_line"
+                },
+            },
+            {
+                "ooni": {
+                    "test_name": "http_header_field_manipulation"
+                },
+            },
+            {
+                "ooni": {
+                    "test_name": "web_connectivity",
+                    "file": "$citizenlab_${probe_cc}_urls"
+                },
+            },
+            {
+                "ooni": {
+                    "test_name": "web_connectivity",
+                    "file": "$citizenlab_global_urls"
+                }
+            }
+        ]
+    }
+    if options["collector"] is not None:
+        deck_data["collector"] = options['collector']
+
+    if options["bouncer"] is not None:
+        deck_data["bouncer"] = options['bouncer']
+
+    deck = NGDeck(deck_data=deck_data)
+    with open(options['output'], 'w+') as fw:
+        deck.write(fw)
+
+    print("Deck written to {0}".format(options['output']))
+    print("Run ooniprobe like so:")
+    print("ooniprobe -i {0}".format(options['output']))
+
+
+@defer.inlineCallbacks
+def get_user_country_code():
+    config.privacy.includecountry = True
+    yield probe_ip.lookup()
```

Comment:
Uhm, would it be possible to pass to `probe_ip.lookup()` a flag to force it to return the country rather than changing a global value?

I am wondering this could break things if, in some way in the future, `config.privacy` is somehow written to disk and hence setting it in this context is going to become persistent.